hack/tests/sanity-check.sh

```diff
@@ -14,6 +14,4 @@ make install
 go run ./hack/generate/samples/generate_all.go
 
 # Make sure repo is still in a clean state.
-# Note that we are ignoring helm manifests with roles.
-# More info: https://github.com/operator-framework/operator-sdk/issues/3873
-git diff --exit-code -- . ':!testdata/helm/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml' ':!testdata/helm/memcached-operator/config/rbac/role.yaml'
```

Comment:
We cannot remove it. See: https://github.com/operator-framework/operator-sdk/issues/3873
Note that if we remove it then will be harder for a developer to contribute to the project since it has diff results when the kind is or not running locally.

Reply:
In this PR, I'm also set the `KUBECONFIG` environment variable during helm generation to ensure that the default static rules are also used.

That way, the samples are always generated without a cluster.